src/app/toolkit/add-tool/components/AddToolHeader.tsx

```diff
@@ -1,15 +1,70 @@
 "use client";
 
 import NavLink from "@/ui/shared/NavLink";
+import Modal from "@/ui/shared/Modal";
 import Spacer from "./Spacer";
 import { ChevronLeftIcon as ChevronLeft } from "@heroicons/react/24/outline";
+import { useAddToolForm } from "@/context/AddToolContext";
+import { useState } from "react";
+import { useRouter } from "next/navigation";
 
 export default function Header() {
+  const [modalOpen, setModalOpen] = useState(false);
+  const router = useRouter();
+  const { formState } = useAddToolForm();
+
+  const hasFormValues = () => {
+    return (
+      formState.name !== "" ||
+      formState.description !== "" ||
+      formState.imageUrl !== "" ||
+      formState.infoUrl !== "" ||
+      formState.categories.length > 0
+    );
+  };
+
+  const handleNavigation = (e: React.MouseEvent) => {
+    e.preventDefault();
+    if (hasFormValues()) {
+      setModalOpen(true);
+    } else {
+      router.push("/toolkit");
+    }
+  };
+
+  const modalForwardButton = {
+    label: "Yes, leave",
+    action: () => {
+      setModalOpen(false);
+      router.push("/toolkit");
+    },
+  };
+
+  const modalBackButton = {
+    label: "No, stay",
+    action: () => setModalOpen(false),
+  };
+
   return (
-    <div className="flex justify-around">
-      <NavLink Icon={ChevronLeft} destination="/toolkit" />
-      <h2 className="text-white text-2xl font-bold">Add Tool</h2>
-      <Spacer />
-    </div>
+    <>
+      <div className="flex justify-around">
+        <NavLink 
+          Icon={ChevronLeft} 
+          destination="/toolkit" 
+          onClick={handleNavigation}
```

Comment:
This seems to be confusing two uses of the navlink - you're both using the destination prop to pass a link that takes the user to that route using the NavLink component, and writing and passing as a prop a handleNavigation function that does the same thing?

I would refrain from using the NavLink component here as the behaviour you're looking for can be achieve with our Button component instead. As you're already calling useRouter in the `AddToolHeader` component, you shouldn't require NavLink to handle that logic for you.

Reply:
I'd far rather use the button component, was using NavLink as that was team consensus when I first made the addTool component.

Duplication of path prop is because:

- I needed to prevent it rerouting before confirmation of modal, which required the handleNavigation
- the preexisting path prop is required and was avoiding breaking everyone else's NavLinks

So I will switch to button yeah?

Reply:
Yes pls !